File: src/scanners/shell.ts
import type { Finding } from "../types";
import { applyCliRules } from "../rules/cli-commands";

/** Scripts shell (.sh, .bash, Dockerfile, Makefile). */
export function scanShell(path: string, content: string): { findings: Finding[]; content: string } {
  return applyCliRules(path, content);
}

File: crates/n2b-scanners/src/shell.rs
// Copyright 2026 aphrody-code
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

use n2b_rules::cli_commands::apply_cli_rules;
use n2b_types::types::Finding;

pub fn scan_shell(path: &str, content: &str) -> (Vec<Finding>, String) {
    apply_cli_rules(path, content)
}
